nntrainer/include/databuffer.h

```diff
@@ -187,17 +169,32 @@ class DataBuffer {
    */
   virtual int clear();
 
+  /**
+   * @brief     set source of data for this given buffer type.
+   * @param[in] type type of buffer.
+   * @param[in] func data provider generator function.
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  virtual int setDataSource(BufferType type, datagen_cb func);
+
+  /**
+   * @brief     set source of data for this given buffer type.
+   * @param[in] type type of buffer.
+   * @param[in] file data provider as file.
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  virtual int setDataSource(BufferType type, std::string file);
+
   /**
    * @brief     get Data from Data Buffer using databuffer param
    * @param[in] BufferType training, validation, test
    * @param[in] outVec feature data ( minibatch size )
    * @param[in] outLabel label data ( minibatch size )
    * @retval    true/false
    */
-  virtual bool getDataFromBuffer(
-    BufferType type,
-    std::vector<std::vector<std::vector<std::vector<float>>>> &out_vec,
-    std::vector<std::vector<std::vector<std::vector<float>>>> &out_label);
+  virtual std::pair<Tensor, Tensor> getDataFromBuffer(BufferType type);
```

Comment:
Need update document.

Reply:
Updated.
As an extra note, this interface allows getting data from generator/file without needing to create multiple copies.